priority/management/commands/calculate-priority.py

```diff
@@ -0,0 +1,121 @@
+        cpu_sum=compute.groupby('Account')['CPUTimeRAW'].sum().astype(int).to_frame(name='cpu_sum')
+        gpu_sum=gpu.groupby('Account')['CPUTimeRAW'].sum().astype(int).to_frame(name='gpu_sum')
+        print(gpu_sum)
+        #generate random ap values for testing
+        #default_Ap=pd.Series(np.random.randint(50000,250000,len(SCW['Ap'])),name='Ap').astype(int)
+        SCW=caculate_AP()
+        #combine the three data frames into one
+        SCW = pd.merge(SCW,cpu_sum, on='Account', how='outer').fillna(0)
+        SCW = pd.merge(SCW,gpu_sum, on='Account', how='outer').fillna(0)
+        #give all new entries an Ap of 50000
+        #SCW.update(default_Ap,overwrite=False,filter_func=new_data)
+        SCW.Ap.loc[SCW.Ap==0]=50000
+        SCW['Cpu_New']=SCW['cpu_sum'].subtract(SCW['CPU_hours'])
+        SCW['Gpu_New']=SCW['gpu_sum'].subtract(SCW['GPU_hours'])
+        
+        
+        #calculate parameter such that K*log(Ap) has a max defined by QOS_levels
+        Max_Ap=SCW['Ap'].max()
+        k=QOS_levels/np.log10(Max_Ap)
+        #check if previous run was prioritezed and if so add the time since last sactdump to total priortized time
+        P=SCW['Qos']>0
+        Prioritsed=SCW[P]
+        Prioritsed['P_CPU'].add(Prioritsed['Cpu_New'])
+        Prioritsed['P_GPU'].add(Prioritsed['Gpu_New'])
+        
+        SCW.update(Prioritsed)
+        SCW['CPU_hours']=SCW['CPU_hours']+SCW['Cpu_New']
+        SCW['GPU_hours']=SCW['GPU_hours']+SCW['Gpu_New']
+        #calculate Priority
+        Priority =(SCW['Ap']-(SCW['P_CPU'] + SCW['P_GPU']*40)).to_frame(name='Priority_Sum').astype(int)
+        Priority['QOS']=SCW['Ap'].map(lambda a : k*np.log10(a)).astype(int)
+        Priority['QOS']= Priority['QOS'].where(Priority['Priority_Sum']>=0,0)
+        Priority['Account']=SCW['Account']
+        
+        
+        #create dataframe for output to django
+        COGS_DB = pd.DataFrame(Priority['Account'], columns=['Account'])
+        COGS_DB['QOS']=Priority['QOS']
+        COGS_DB['CPU_hours']=SCW['CPU_hours']
+        COGS_DB['GPU_hours']=SCW['GPU_hours']
+        COGS_DB['P_CPU']=SCW['P_CPU']
+        COGS_DB['P_GPU']=SCW['P_GPU']
+        COGS_DB['Ap']=SCW['Ap']
+        COGS_DB['Date']=pd.date_range(start=Today, end=Today,periods=len(SCW.index))
+        #output Acoount and QOS to pipe seperated csv file 
+        Priority.to_csv(path_or_buf='~/Documents/Cogs3/cogs3/priority/fixtures/QOS_output.csv',sep='|',columns=['Account','QOS'],index=False)
+  #chack put in place to prevent writing data twice on the same day
+        assert (not slurm_priority.objects.filter(Date__year=Today.year, Date__month=Today.month, Date__day=Today.day).exists()),'Data for today already exists in django database'
+        # Write data to django database
+        COGS_DB.to_sql('priority_slurm_priority',index=False,con=engine,if_exists='append')
+        engine.execute("SELECT * FROM priority_slurm_priority").fetchall()
```

Comment:
This line seems to execute a query and then discard the results. Is this the intention?